docs/source/posts/2024/2024-08-21-final-report-robin.rst

```diff
@@ -0,0 +1,305 @@
+.. image:: https://developers.google.com/open-source/gsoc/resources/downloads/GSoC-logo-horizontal.svg
+   :height: 40
+   :target: https://summerofcode.withgoogle.com/programs/2023/projects/ED0203De
+
+.. image:: https://www.python.org/static/img/python-logo@2x.png
+   :height: 40
+   :target: https://summerofcode.withgoogle.com/programs/2023/organizations/python-software-foundation
+
+.. image:: https://python-gsoc.org/logos/fury_logo.png
+   :width: 40
+   :target: https://fury.gl/latest/index.html
+
+
+
+Google Summer of Code Final Work Product
+========================================
+
+.. post:: August 21 2024
+    :author: Robin Roy
+    :tags: google
+    :category: gsoc
+
+- **Name:** `Robin Roy <https://github.com/robinroy03>`_
+- **Organization:** Python Software Foundation
+- **Sub-Organization:** FURY
+- **Project:** `Improving Community Engagement: AI communication automation using LLM <https://github.com/fury-gl/fury/wiki/Google-Summer-of-Code-2024-(GSOC2024)#project-2-improving-community-engagement-ai-communication-automation-using-llm>`_
+
+
+Abstract
+--------
+
+The goal of this project was to implement a `Large Language Model(LLM) <https://developers.google.com/machine-learning/resources/intro-llms>`_ chatbot that understands the FURY repository. The purpose of the project is to reduce the barrier of entry to scientific visualization. `Retrieval Augmented Generation(RAG) <https://www.pinecone.io/learn/retrieval-augmented-generation/>`_ was used to get the necessary context for every user query. Multiple variations were explored, including Fine-Tuning models, mixing Fine-Tuning and RAG and RAG alone. Multiple `chunking strategies <https://github.com/FullStackRetrieval-com/RetrievalTutorials/blob/main/tutorials/LevelsOfTextSplitting/5_Levels_Of_Text_Splitting.ipynb>`_ were also explored for data collection and storage. The models are served to the user through a Discord Bot and a GitHub App. All the API endpoints are hosted using `HuggingFace Spaces <https://huggingface.co/robinroy03>`_. `Pinecone <https://www.pinecone.io/>`_ was used as the database for storing embeddings. Benchmarking, data collection, and testing were done on `another repository <https://github.com/robinroy03/FURY-data-script>`_.
```

Comment:
I didn't know about your testing repository. Will take a look into it, thanks for including it. It vital to include where your work is located. Thanks